src/Searchable/Aggregators.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of Scout Extended.
+ *
+ * (c) Algolia Team <contact@algolia.com>
+ *
+ *  For the full copyright and license information, please view the LICENSE
+ *  file that was distributed with this source code.
+ */
+
+namespace Algolia\ScoutExtended\Searchable;
+
+class Aggregators extends Aggregator
+{
+    /**
+     * Boot multiple aggregators.
+     *
+     * @return void
+     */
+    public static function bootSearchables(array $searchables): void
+    {
+        (new static)->registerSearchableMacros();
+
+        $models = [];
+
+        foreach ($searchables as $searchable) {
+            foreach ((new $searchable)->getModels() as $model) {
+                $models[$model][] = $searchable;
```

Comment:
We need to force this to a string to prevent PHPStan errors in the next `foreach` loop: the tool thinks the array key there can be `string|int`.
```suggestion
                $models[(string) $model][] = $searchable;
```